test/utils/executor.hpp

```diff
@@ -119,16 +93,17 @@ inline void init_executor(std::shared_ptr<gko::ReferenceExecutor> ref,
     if (gko::HipExecutor::get_num_devices() == 0) {
         throw std::runtime_error{"No suitable HIP devices"};
     }
-    exec = gko::HipExecutor::create(0, ref, false, gko::default_hip_alloc_mode,
-                                    stream);
+    exec = gko::HipExecutor::create(ResourceEnvironment::hip_device_id, ref,
+                                    false, gko::default_hip_alloc_mode, stream);
 }
 
 
 inline void init_executor(std::shared_ptr<gko::ReferenceExecutor> ref,
                           std::shared_ptr<gko::DpcppExecutor>& exec)
 {
     if (gko::DpcppExecutor::get_num_devices("gpu") > 0) {
-        exec = gko::DpcppExecutor::create(0, ref, "gpu");
+        exec = gko::DpcppExecutor::create(ResourceEnvironment::sycl_device_id,
+                                          ref, "gpu");
```

Comment:
I guess we just don't run SYCL CPU tests in parallel, right? Controlling their parallelism seems hard

Reply:
I just played around with SYCL's create_sub_devices a bit, and it seems like this might actually not be that hard.
```cpp
sycl::device d = sycl::cpu_selector().select_device();
std::cout << d.get_info<sycl::info::device::max_compute_units>() << std::endl;
auto devices = d.create_sub_devices<sycl::info::partition_property::partition_equally>(4);
for (auto dev : devices) {
    std::cout << dev.get_info<sycl::info::device::max_compute_units>() << std::endl;
}
```
The only difference is that OpenCL actually binds to specific cores, while OpenMP doesn't give such fine control

Reply:
In the latest version, we just have a single SYCL device that may be CPU or GPU, allowing limited oversubscription. That should be sufficient.